---
title: "spike_correspondence_figure1b"
format: gfm
toc: true
theme: cosmo
---

```{r}
library(tidyverse)
library(RColorBrewer)
theme_set(theme_classic())
library(DescTools)
```
# Our dataset

Sequenced on NextSeq 550, demultiplexed with Illumina bcl2fastq

## Trimming .FASTQ Files

```{bash, eval = FALSE}
base=$(basename $i _R1_001.fastq.gz)
trimmomatic SE ${base}_R1_001.fastq.gz ${dir}/${base}.trim.fastq.gz \
-threads 8 \
ILLUMINACLIP:/trimmomatic-0.39-hdfd78af_2/share/trimmomatic-0.39-2/adapters/TruSeq3-SE.fa:2:30:7 \
LEADING:3 TRAILING:3 SLIDINGWINDOW:4:20 MINLEN:36
```


## Alignment: .FASTQ -> .SAM

### Genome Preparation: 

Genomes Downloaded with HOMER: hg38, dm6, sacCer3

```{bash, eval = FALSE}
perl /gpfs/data01/bennerlab/path/to/homer/configureHomer.pl -install hg38

perl /gpfs/data01/bennerlab/path/to/homer/configureHomer.pl -install dm6

perl /gpfs/data01/bennerlab/path/to/homer/configureHomer.pl -install sacCer3
```

Add chromosome suffixes to identify spike-in chromosomes

Fasta file of genome location: /homer/data/genomes/hg38/genome.fa

```{bash, eval = FALSE}
## Fly dm6 Genome
# suffix added "_dm6"
sed 's/>.*/&_dm6/' genome.fa > genome_dm6.fa

# check by printing fastq headers
perl -ne 'if(/^>(\S+)/){print "$1\n"}' genome_dm6.fa

## Yeast sacCer3 Genome
# suffix added "_sac3"
sed 's/>.*/&_sac3/' genome.fa > genome_sac3.fa

# check by printing fastq headers
perl -ne 'if(/^>(\S+)/){print "$1\n"}' genome_sac3.fa
```

Combine spike-in/target genomes: 

```{bash, eval = FALSE}
cat ${dir}/dm6/genome_dm6.fa ${dir}/sacCer3/genome_sac3.fa ${dir}/hg38/genome.fa > genome_hg38_dm6_sac3.fa
```

Index genome: 
```{bash, eval = FALSE}
bwa index -p hg38_dm6_sac3 genome_hg38_dm6_sac3.fa
```

Alignment:
```{bash, eval = FALSE}
bwa mem -t 4 ~/data/genome_index/genome_prefix ${file}.fastq > ${file}.sam
```

## Remove PCR Duplicates

```{bash, eval = FALSE}
# single end reads: don't need to collate
# fixmate can go from sam to bam 
samtools fixmate -m ${file}.sam ${file}.fixmate.bam
samtools sort ${file}.fixmate.bam -o ${file}.sorted.bam
samtools markdup -r -s ${file}.sorted.bam ${file}.nodup.bam
```

## Separate Alignment File into Species-specific Alignments

Split alignment file into files for each chromosome
```{bash, eval = FALSE}
bam splitChromosome --in ${file}.nodup.bam --out ${file}.
```

Remove *chrUn*, *random* files

Merge chromosome files to get one alignment file per species. Need to merge spike-in species first, then remove those chromosomes

```{bash, eval = FALSE}
# create dm6 spike-in alignment file
samtools merge ${file}.dm6.bam ${file}.chr*_dm6.bam

# remove dm6 spike-in chromosome files
rm ${file}.chr*_dm6.bam

# create sac3 spike-in alignment file
samtools merge ${file}.sac3.bam ${file}.chr*_sac3.bam

# remove sac3 spike-in chromosome files
rm ${file}.chr*_sac3.bam

# create target alignment file
samtools merge ${file}.hg38.bam ${file}.chr*.bam
```

Convert BAM to SAM (-h to keep header), remove suffixes from files with sed.

```{bash, eval = FALSE}
# remove first set of spike-in suffixes
samtools view -h ${file}.bam | sed -e 's/\_dm6//g' > ${file}.nosuffix.sam

# remove second set of spike-in suffixes
samtools view -h ${file}.nosuffix.sam | sed -e 's/\_sac3//g' > ${file}.nosuffix2.sam
```

## Make HOMER Tag Directories

By default, HOMER `makeTagDirectory` or `batchMakeTagDirectory` only keep primary alignments with MAPQ > 10. Tag directories are also read-depth normalized to 10 million reads unless otherwise specified. To create Tag Directories in batch mode, first create a tsv file called a tagkey, containing the names for each input file and each desired output tag directory.

Format of the tagkey file: TSV  <br>
file1-tagdir   \t> file1.nosuffix2.sam <br>
file2-tagdir   \t> file2.nosuffix2.sam <br>
file3-tagdir   \t> file3.nosuffix2.sam <br>

Then run HOMER `batchMakeTagDirectory.pl` for each species. Example with hg38:

```{bash, eval = FALSE}
batchMakeTagDirectory.pl tagkey -genome hg38 \
-cpu 8 -fragLength 150
```

Repeat for dm6, sacCer3 genomes

## Visualize with BigWigs

Can make BigWigs from bam file with Deeptools, or from Tag Directories with HOMER `makeBigWig.pl`

```{bash, eval = FALSE}
makeBigWig.pl ${file}-tagdir/ hg38 \
-webdir /path/to/webdirectory -url http://webdirectoryurl/
```

Note: you can also make BedGraphs with HOMER `makeBedGraph.pl`

Repeat for dm6, sacCer3 genomes

## Determine Normalization Factor

## Analysis with HOMER

### Histograms at TSS

Recommended parameters for H3K79me2: Make histogram of size 4kb centered at TSS, with bin size 25bp.

```{bash, eval = FALSE}
annotatePeaks.pl tss hg38 -size 4000 -hist 25 \
-d ${file}1-tagdir ${file}2-tagdir \
> histogram_tss_hg38_samples.txt
```

### Peak Finding

Recommended parameters for human histone mark peak finding: -style histone, -size 1000, -minDist 2500. Note: yeast genome is gene/acetylation dense, so if peak finding in yeast is desired, paramters should be tweaked.

```{bash, eval = FALSE}
findPeaks.pl ${file}-tagdir -style histone -size 1000 -minDist 2500 \
-i ${file}-input-tagdir > ${file}.regions.txt
```

### Quantification at Peaks

```{bash, eval = FALSE}
annotatePeaks.pl ${file}.regions.txt -size 1000 \
-d ${file1}-tagdir ${file2}-tagdir > counts_regions_1kb_samples.txt
```

## Megapeak: Get H3K9ac regions to quantify signal

1. Combined all samples to one "mega-ChIP-seq sample", and all inputs to one "mega-input control". 
2. Did peak finding with "mega ChIP-seq sample" with "mega-input control" as background. The overall increased read-depth of sample and input files will increase the tag threshold for peak finding, resulting in more significant peaks.
3. Quantified H3K9ac signal at these peaks for every sample, making a histogram of size 4000 and bin size 25.

```{bash, eval = FALSE}
findPeaks.pl ${file}.megasample-tagdir -style histone -size 1000 -minDist 2500 \
-i ${file}.megainput-tagdir > megasample_H3K9ac.regions.txt

annotatePeaks.pl megasample_H3K9ac.regions.txt hg38 -size 4000 -hist 25 \
-d ${file1}-tagdir ${file2}-tagdir ${file3}-tagdir .... > hist_K9ac_hg38_allsamples_LP78.txt
```

# Import histogram data

```{r}
hist_K9ac_allsamples_LP78 <- read.delim("~/Research/LP_78/hist_K9ac_hg38_allsamples_LP78.txt")
```

```{r}
process_my_histograms <- function(x, .x) {
    colnames(x)[1] <- "Distance_from_center"
    x <- x %>% 
    rename_with(~ gsub(".hg38.tagdir", "", .x), contains("tagdir")) %>% 
    #rename_with(~ gsub(".+\\_Hela_", "Hela_", .x), contains("Hela")) %>%
    rename_with(~ gsub("\\.[[:digit:]]$", "_minus", .x), contains("Tags")) %>% 
    rename_with(~ gsub("\\.\\.\\.", "_", .x), contains("Tags"))
    
    xcov <- x %>% select(contains("Coverage"))
    xcov$Distance_from_center <- x$Distance_from_center
    
    xcovlong <- 
    xcov %>% pivot_longer(
      cols = -"Distance_from_center", 
      names_to = "Sample", 
      values_to = "Coverage")
  }
```

```{r}
hist_K9ac_allsamples_LP78_tidy <- 
  process_my_histograms(hist_K9ac_allsamples_LP78)
```

```{r}
hist_K9ac_allsamples_LP78_sepIP <- hist_K9ac_allsamples_LP78_tidy[grep("K9ac", hist_K9ac_allsamples_LP78_tidy$Sample), ] %>% separate_wider_regex(cols = Sample, patterns = c(
  "X", 
  id = "78[:lower:]",
  "\\_",
  cell = "[:alpha:]+", 
  "\\_", 
  ratio = ".+", 
  "\\_", 
  antibody = ".+", 
  "\\_", 
  replicate = ".+", 
  ".Coverage"))
```

```{r, fig.height = 5, fig.width = 6}
ggplot(data = hist_K9ac_allsamples_LP78_sepIP) + 
  aes(x = Distance_from_center, y = Coverage, group=interaction(ratio, antibody, replicate), color = ratio) + 
  geom_line(alpha = 0.9, linewidth = 1.1) + 
  labs(title = "Hela H3K9ac megasample peaks", 
       x = "Distance from Peak Center") +
  scale_color_manual(
    values = colorRampPalette(brewer.pal(9, "Blues"))(8)[3:8],
    name = "Cell Ratio", 
    labels = c("0% interphase", "25% interphase", "50% interphase", "75% interphase", "95% interphase", "100% interphase")) + 
  theme_classic() + 
  theme(legend.position = c(0.84, 0.76), 
                          legend.background = element_rect(
                                  size=0.7, linetype="solid", 
                                  colour ="grey20")) 
```

### Calculate Signal at Peaks: Area under Histogram Curve

```{r}
samples <- unique(hist_K9ac_allsamples_LP78_sepIP$id)
colnames(hist_K9ac_allsamples_LP78)[1] <- "Distance_from_center"
x <- hist_K9ac_allsamples_LP78$Distance_from_center
AUC_peaks <- matrix(data = "", nrow = length(samples), ncol = 1)
AUC_peaks <- data.frame(AUC_peaks, row.names = samples)

for (i in 1:length(samples)) {

  y <- hist_K9ac_allsamples_LP78_sepIP %>% 
    filter(id == samples[i]) %>%
    select(Coverage)
  y <- pull(y, Coverage)

AUC_peaks[i, ] <- AUC(x, y, method = c("trapezoid"))

}
```

Processing Signal at peaks, add expected line
```{r}
AUC_peaks$AUC_peaks <- as.numeric(AUC_peaks$AUC_peaks)

interphase <- rep(c(0, 25, 50, 75, 95, 100), each = 3)
AUC_peaks$interphase <- as.numeric(interphase)
```

 Scale points from 0-1

```{r}
avg_0inter <- mean(c(AUC_peaks[1,1], AUC_peaks[2,1], AUC_peaks[3,1]))
avg_25inter <- mean(c(AUC_peaks[4,1], AUC_peaks[5,1], AUC_peaks[6,1]))
avg_50inter <- mean(c(AUC_peaks[7,1], AUC_peaks[8,1], AUC_peaks[9,1]))
avg_75inter <- mean(c(AUC_peaks[10,1], AUC_peaks[11,1], AUC_peaks[12,1]))
avg_95inter <- mean(c(AUC_peaks[13,1], AUC_peaks[14,1], AUC_peaks[15,1]))
avg_100inter <- mean(c(AUC_peaks[16,1], AUC_peaks[17,1], AUC_peaks[18,1]))
```

```{r}
AUC_peaks <- AUC_peaks %>%
  mutate(minmaxnorm = (AUC_peaks-avg_0inter)/(avg_100inter-avg_0inter) )
```

```{r}
minmaxnorm_avg_0inter <- mean(c(AUC_peaks[1,3], AUC_peaks[2,3], AUC_peaks[3,3]))
minmaxnorm_avg_25inter <- mean(c(AUC_peaks[4,3], AUC_peaks[5,3], AUC_peaks[6,3]))
minmaxnorm_avg_50inter <- mean(c(AUC_peaks[7,3], AUC_peaks[8,3], AUC_peaks[9,3]))
minmaxnorm_avg_75inter <- mean(c(AUC_peaks[10,3], AUC_peaks[11,3], AUC_peaks[12,3]))
minmaxnorm_avg_95inter <- mean(c(AUC_peaks[13,3], AUC_peaks[14,3], AUC_peaks[15,3]))
minmaxnorm_avg_100inter <- mean(c(AUC_peaks[16,3], AUC_peaks[17,3], AUC_peaks[18,3]))
```

### Make line of expected signal: 

```{r}
observed_line <- c(minmaxnorm_avg_100inter, 
                   minmaxnorm_avg_95inter, 
                   minmaxnorm_avg_75inter,  
                   minmaxnorm_avg_50inter, 
                   minmaxnorm_avg_25inter,  
                   minmaxnorm_avg_0inter)

expected_line <- c(0, 0.25, 0.5, 0.75, 0.95, 1)

percent_inter_mean <- rep(c(0, 25, 50, 75, 95, 100))

percent_mit_mean <- rep(c(100, 75, 50, 25, 5, 0))
AUC_peaks$mitotic <- rep(percent_mit_mean, each = 3)

observed_vs_expected_LP78 <- data.frame(cbind(
  percent_inter_mean, percent_mit_mean, 
  expected_line, observed_line))
```

### Plot Read normalized signal vs expected signal: Fig 1b

```{r, fig.height = 4.5, fig.width = 4.5}
ggplot() +
  geom_point(data = AUC_peaks, 
             aes(x = as.numeric(mitotic), y = minmaxnorm), 
             size = 2, alpha = 0.7, shape = 3, color = "grey30", stroke = 3) +
  scale_color_manual(name = "Cell Ratio", 
    labels = c("0% interphase", "25% interphase", "50% interphase", "75% interphase", "95% interphase", "100% interphase")) +
  geom_line(data = observed_vs_expected_LP78, 
            aes(x = as.numeric(percent_mit_mean), y = expected_line), linewidth = 1.1, color = "grey30") +
 labs(title = "Relative H3K9ac Signal", 
       subtitle = "Read Normalized, scaled 0-1",
       x = "% Mitotic Cells", 
       y = "Relative H3K9ac Signal") + 
  theme(legend.position = "none")
```

Calculated Rsquared

```{r}
get_Rsquared <- function(AUC_peaks) {
  AUC_peaks$expected <- rep(c(0, 0.25, 0.5, 0.75, 0.95, 1), each = 3)
  
  AUC_peaks$error <- AUC_peaks$minmaxnorm - AUC_peaks$expected
  
  mse = sum(((AUC_peaks$error)^2)/3)
  
  SSres = sum((AUC_peaks$error)^2)
  meanobserved <- mean(AUC_peaks$minmaxnorm)
  SStotal = sum((AUC_peaks$minmaxnorm-meanobserved)^2)
  rsquared = 1 - (SSres/SStotal)
  rsquared
}
```

```{r}
get_Rsquared(AUC_peaks)
```

